### PR TITLE
[Feature] Enhancements for afk.

### PR DESCRIPTION
**Pull request by Dolphin2Point1**
This is for #1928. It added cancelAfkOnChat, cancelAfkOnDisconnect, afkBroadcastOnDisconnect, & afkBroadcastOnMutedChat. afkBroadcastOnDisconnect & afkBroadcastOnMutedChat may be a little unnecessary, but I put them in anyway.

### CODE REVIEW

**Comment by Dolphin2Point1** [comment]
Whoops. Got quite a few errors in compiling. Let me fix that.

**Comment by Dolphin2Point1** [comment]
[Click here for the code](https://github.com/loper12/Essentials/tree/AFK-Enhancements). BTW I fixed it... Hopefully.

**Comment by Dolphin2Point1** [comment]
Watch out. I'm drowning in compilation errors. I will fix them.

**Comment by Dolphin2Point1** [comment]
There. Now it should work.

**Comment by mdcfe** [comment]
Could you fix the diff?

**Comment by Dolphin2Point1** [comment]
I reverted it because I didn't want the versions to be different. Maybe I can hook on the commit for the new one.

**Comment by mdcfe** [comment]
I don't think there's any reason to have separate messages to be broadcast when AFK is disabled on disconnect or chat. In either case, it's plainly obvious that the user is no longer AFK because they'll have broadcasted something in chat for everyone to see.

**Comment by Dolphin2Point1** [comment]
I still wanted it just in case. The automatic setting is false.

File: Essentials/src/config.yml [hunk 1]
```diff
@@ -396,6 +396,23 @@ cancel-afk-on-interact: true
...
+# This option will tell the players when the afk players disconnect, but only if cancel-afk-on-disconnect
+# is true.
+# This option is automatically false because people would probably know that they are not afk.
+broadcast-afk-on-disconnect: false
+
+# This setting tells the players when an afk player chats, but it only works if cancel-afk-on-chat is
+# set to true.
+# This option is automatically false because again, it's really obvious.
+broadcast-afk-on-chat: false
+
```
Comment (Dolphin2Point1) [inline Essentials/src/config.yml:415]: Tell me what you think about this part of the config I added.

**Comment by Dolphin2Point1** [comment]
Imagine they had separate channels for chat. It wouldn't be super obvious.

**Comment by mdcfe** [comment]
Thanks for the contribution. However, in it's current state, this PR adds lots of unnecessary features with limited utility. In addition, resetting AFK on logout shouldn't be a feature you can turn on and off - #1928 is a bug.

I'm going to close this PR in favour of #2458, which fixes this in a much cleaner way without extra bloat or feature creep.

**Pull request closed without merging**